docs/dev-guide/python/versioning.md

```diff
@@ -120,3 +142,178 @@ await workflow.execute_activity(
 )
 # ...
 ```
+
+## How to use the Python SDK Patching API {#python-sdk-patching-api}
+
+<!-- DO NOT EDIT THIS FILE DIRECTLY.
+THIS FILE IS GENERATED from https://github.com/temporalio/documentation-samples-python/blob/edu-1171/version_your_workflows/workflow_1_initial_dacx.py. -->
+
+In principle, the Python SDK's patching mechanism operates similarly to other SDKs in a "feature-flag" fashion. However, the "versioning" API now uses the concept of "patching in" code.
+
+To understand this, you can break it down into three steps, which reflect three stages of migration:
+
+- Running `pre_patch_activity` code while concurrently patching in `post_patch_activity`.
+- Running `post_patch_activity` code with deprecation markers for `my-patch` patches.
+- Running only the `post_patch_activity` code.
+
+Let's walk through this process in sequence.
+
+Suppose you have an initial Workflow version called `pre_patch_activity`:
+
+<a class="dacx-source-link" href="https://github.com/temporalio/documentation-samples-python/blob/edu-1171/version_your_workflows/workflow_1_initial_dacx.py">View source code</a>
+
+```python
+from datetime import timedelta
+
+from temporalio import workflow
+
+with workflow.unsafe.imports_passed_through():
+    from activities import pre_patch_activity
+# ...
+@workflow.defn
+class MyWorkflow:
+    @workflow.run
+    async def run(self) -> None:
+        self._result = await workflow.execute_activity(
+            pre_patch_activity,
+            schedule_to_close_timeout=timedelta(minutes=5),
+        )
+
+    @workflow.query
+    def result(self) -> str:
+        return self._result
+```
+
+<!-- DO NOT EDIT THIS FILE DIRECTLY.
+THIS FILE IS GENERATED from https://github.com/temporalio/documentation-samples-python/blob/edu-1171/version_your_workflows/workflow_4_patch_complete_dacx.py. -->
+
+Now, you want to update your code to run `post_patch_activity` instead. This represents your desired end state.
+
+<a class="dacx-source-link" href="https://github.com/temporalio/documentation-samples-python/blob/edu-1171/version_your_workflows/workflow_4_patch_complete_dacx.py">View source code</a>
+
+```python
+from datetime import timedelta
+
+from temporalio import workflow
+
+with workflow.unsafe.imports_passed_through():
+    from activities import post_patch_activity
+# ...
+@workflow.defn
+class MyWorkflow:
+    @workflow.run
+    async def run(self) -> None:
+        self._result = await workflow.execute_activity(
+            post_patch_activity,
+            schedule_to_close_timeout=timedelta(minutes=5),
+        )
+
+    @workflow.query
+    def result(self) -> str:
+        return self._result
+```
+
+<!-- DO NOT EDIT THIS FILE DIRECTLY.
+THIS FILE IS GENERATED from https://github.com/temporalio/documentation-samples-python/blob/edu-1171/version_your_workflows/workflow_4_patch_complete_dacx.py. -->
+
```

Comment:
Maybe add a why here. Why can't I do this? It can be as simple as

> ...using the `pre_patch_activity` code, otherwise you are likely to cause a non-deterministic error.